docs/how_tos/feedback.rst

```diff
@@ -0,0 +1,409 @@
+.. _payment/data/handleRequestError.js: https://github.com/openedx/frontend-app-payment/blob/master/src/payment/data/handleRequestError.js
+.. _payment/data/sagas.js: https://github.com/openedx/frontend-app-payment/blob/1d631c51035eb8405ce9b03e0fa64a5a6e386268/src/payment/data/sagas.js
+.. _payment/AlertCodeMessages.jsx: https://github.com/openedx/frontend-app-payment/blob/1d631c51035eb8405ce9b03e0fa64a5a6e386268/src/payment/AlertCodeMessages.jsx
+.. _components/formatted-alert-list/FormattedAlertList.jsx: https://github.com/openedx/frontend-app-payment/blob/1d631c51035eb8405ce9b03e0fa64a5a6e386268/src/components/formatted-alert-list/FormattedAlertList.jsx
+
+
+What is the fallback error?
+---------------------------
+
+If a saga calls `handleErrors()`_ but the API response is not in one of the
+formats above or ``error_code`` is ``fallback-error``, ``handleErrors()`` will
+throw up the `FallbackErrorMessage`_.
+
+.. _FallbackErrorMessage: https://github.com/openedx/frontend-app-payment/blob/master/src/feedback/FallbackErrorMessage.jsx
+.. _handleErrors(): https://github.com/openedx/frontend-app-payment/blob/1d631c51035eb8405ce9b03e0fa64a5a6e386268/src/feedback/data/sagas.js#L10C22-L10C22
+
+
+When are snake_case dictionary keys auto-converted to camelCase?
+----------------------------------------------------------------
+
+The payment module's `payment/data/service.js`_ (also dubbed the
+`PaymentApiService`_ in payment/data/sagas.js) funnels many but not all API
+responses through `transformResults`_, which recursively converts snake_case
+dictionary keys of the response into camelCase.
+
+This was done because the convention for Python backends generating the API
+responses is to use snake_case, while the convention for JavaScript frontends
+processing the API responses is to use camelCase.
+
+See `payment/data/service.js`_ for if and when `transformResults`_ is used.
+
+`payment/data/handleRequestError.js`_ also converts the top-level keys
+``error_code`` and ``user_message`` and ``message_type`` into camelCase.
+
+.. _PaymentApiService: https://github.com/openedx/frontend-app-payment/blob/1d631c51035eb8405ce9b03e0fa64a5a6e386268/src/payment/data/sagas.js#L34
+.. _transformResults: https://github.com/openedx/frontend-app-payment/blob/1d631c51035eb8405ce9b03e0fa64a5a6e386268/src/payment/data/utils.js#L158-L172
+
+
+What about internationalization?
+--------------------------------
+
+Prefer using ``error_code``. It uses a pre-built component which should support
+internationalization like any other component with translatable strings in
+frontend-app-payment.
+
+
+For more information
+--------------------
+
+See `Feedback module's README and components`_, and especially the
+implementation in `feedback/data/sagas.js`_.
+
+.. _Feedback module's README and components: https://github.com/openedx/frontend-app-payment/tree/master/src/feedback
+
+
+Future work
```

Comment:
Are there open tickets for these items? if not should we create them and link them here? (in maintenance?)

Reply:
Good point! Created: [REV-3618](https://2u-internal.atlassian.net/browse/REV-3618)